src/main/java/uk/gov/pay/directdebit/payments/resources/PaymentResource.java

```diff
@@ -57,6 +58,10 @@ public Response collectPaymentFromMandate(@PathParam("accountId") GatewayAccount
         LOGGER.info("Received collect payment from mandate request");
         collectPaymentRequestValidator.validate(collectPaymentRequestMap);
         Payment paymentToCollect = collectService.collect(gatewayAccount, CollectPaymentRequest.of(collectPaymentRequestMap));
+        if(!paymentService.isPaymentMandateStateValid(paymentToCollect)) {
```

Comment:
can you auto-format please

Reply:
this should be done inside the collect service I think. we definitely shouldn't be doing it after we collect the payment.

Reply:
Refactored